File: mobile-app/lib/models/note_template.dart
class NoteTemplate {
  final String? id;
  final String name;
  final String? description;
  final String content;
  final String category;
  final bool isPublic;
  final String? createdBy;
  final int usageCount;
  final DateTime createdAt;

  NoteTemplate({
    this.id,
    required this.name,
    this.description,
    required this.content,
    this.category = 'general',
    this.isPublic = false,
    this.createdBy,
    this.usageCount = 0,
    required this.createdAt,
  });

  factory NoteTemplate.fromJson(Map<String, dynamic> json) {
    return NoteTemplate(
      id: json['_id'] ?? json['id'],
      name: json['name'] ?? '',
      description: json['description'],
      content: json['content'] ?? '',
      category: json['category'] ?? 'general',
      isPublic: json['is_public'] ?? false,
      createdBy: json['created_by']?['_id'] ?? json['created_by'],
      usageCount: json['usage_count'] ?? 0,
      createdAt: DateTime.parse(json['created_at'] ?? json['createdAt']),
    );
  }
}

File: backend/models/NoteTemplate.js
/**
 * Modèle pour les templates de notes
 */
const mongoose = require('mongoose');

const noteTemplateSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    trim: true,
    maxlength: 100,
  },
  description: {
    type: String,
    maxlength: 500,
  },
  content: {
    type: String,
    default: '',
  },
  category: {
    type: String,
    default: 'general',
    enum: ['general', 'meeting', 'project', 'personal', 'work', 'education'],
  },
  is_public: {
    type: Boolean,
    default: false,
  },
  created_by: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    default: null,
  },
  usage_count: {
    type: Number,
    default: 0,
  },
  created_at: {
    type: Date,
    default: Date.now,
    index: true,
  },
}, {
  timestamps: true,
});

// Index pour les requêtes fréquentes
noteTemplateSchema.index({ category: 1, is_public: 1 });
noteTemplateSchema.index({ created_by: 1 });
noteTemplateSchema.index({ usage_count: -1 });

const NoteTemplate = mongoose.model('NoteTemplate', noteTemplateSchema);

module.exports = NoteTemplate;





